src/osd/PG.cc

```diff
@@ -2646,37 +2644,36 @@ void PG::_update_calc_stats()
       //   - not in up  Compute misplaced objects excluding num_missing
       // backfill       Compute total objects already backfilled
       if (!is_backfill_targets(p)) {
-        unsigned osd_missing;
+        unsigned osd_missing, osd_objects;
         // primary handling
         if (p == pg_whoami) {
           osd_missing = pg_log.get_missing().num_missing();
           info.stats.stats.sum.num_objects_missing_on_primary =
               osd_missing;
-          object_copies += num_objects; // My local (primary) count
         } else {
           assert(peer_missing.count(p));
           osd_missing = peer_missing[p].num_missing();
-          object_copies += peer_info[p].stats.stats.sum.num_objects;
         }
-        missing += osd_missing;
+
+        osd_objects = MAX(0, num_objects - osd_missing);
```

Comment:
please note, `num_objects` will be promoted to `unsigned` when evaluating this expression which involves `unsigned` operand. so
`MAX(0, num_objects - osd_missing)` is always evaluated to `num_objects - osd_missing`. and the latter could be an underflow of unsigned integer.

Reply:
@tchaikov I tested this and it shows that the results of (int64_t)num_objects - (unsigned)osd_missing is signed (int64_t).  So the MAX comparison to 0 will work and then final results changed back to unsigned.

Reply:
@dzafman ahh, right! i see. i didn't realize that `num_objects` is an `int64_t`. the size of `int64_t` is greater than that of `unsigned`, hence has higher integer conversion rank. so `(int64_t)num_objects - (unsigned)osd_missing` is a signed (int64_t).